zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -77,6 +77,11 @@ sealed trait Http[-R, +E, -A, +B] extends (A => ZIO[R, Option[E], B]) { self =>
    */
   final def apply(a: A): ZIO[R, Option[E], B] = execute(a).toZIO
 
+  /**
+   * Applies Http based only if the condition function evaluates to true
+   */
+  final def when[R1 <: R, E1 >: E, A2 <: A, B1 >: B](f: A2 => Boolean): Http[R, E, A2, B] = Http.When(f, self)
+
```

Comment:
Add HttpApp specific operators like `whenPath`, `whenHeaders` and `whenRequest`

Reply:
We aren't using all the type params R1, E1 etc., why have them?

Reply:
We should have two overloads for `whenPath` one should take a `Path` and one should take a `String`.